src/service_data_structure.cc

```diff
@@ -201,6 +201,28 @@ ServiceDataStructure::ReturnCodes ServiceDataStructure::UserSession::PostChirp(
     return INTERNAL_BACKEND_ERROR;
   }
 
+
+
+  // Parse the chirp text to find any tags
+  std::string::size_type start = text.find('#');
```

Comment:
I would recommend moving the string parsing to a helper.

Reply:
Fixed.